### PR TITLE
[MERGE-ATTN](function) fix str_to_date default return type scale for nereids

### PR DESCRIPTION
**Pull request by zclllyybb**
# If merge this pr into other branch, must together with https://github.com/apache/doris/pull/25707

## Proposed changes

Issue Number: close #xxx

when we have return type of DatetimeV2, will set its' scale to MAX, which is 6.
btw, fix the wrong result of format string parsing when there's `%f` and also other things trailing.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

File: gensrc/script/doris_builtins_functions.py [hunk 1]
```diff
@@ -1104,6 +1102,10 @@
         [['datediff'], 'INT', ['DATEV2', 'DATEV2'], ''],
         [['timediff'], 'TIMEV2', ['DATEV2', 'DATEV2'], ''],
 
+        [['str_to_date'], 'DATETIMEV2', ['VARCHAR', 'VARCHAR'], 'ALWAYS_NULLABLE'],
+        [['str_to_date'], 'DATETIMEV2', ['STRING', 'STRING'], 'ALWAYS_NULLABLE'],
+        [['str_to_date'], 'DATEV2', ['VARCHAR', 'VARCHAR'], 'ALWAYS_NULLABLE'],
```
Comment (HappenLee) [inline gensrc/script/doris_builtins_functions.py:1107]: remov the unless register function

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Comment by zclllyybb** [comment]
这个pr涉及平滑升级，只能等到2.1再合入

**Pull request merged**